factor_analyzer/factor_analyzer.py

```diff
@@ -114,6 +116,70 @@ def calculate_bartlett_sphericity(x):
     return statistic, p_value
 
 
+def calculate_cng_indices(
+    data: np.ndarray, model: str = "components"
+) -> Tuple[int, pd.DataFrame]:
+    """Calculate the Cattel-Nelson-Gorsuch indices, which are used to determine
+    the appropriate number of factors for a factor analysis.
+
+    Direct port of nCng function from nFactors package:
+    https://rdrr.io/cran/nFactors/man/nCng.html
+
+    Parameters
+    ----------
+    data : array-like
+        The array of samples x observable for which to calculate CNG indices
+    model : str
+        "components" or "factors"
+
+    Returns
+    -------
+    num_factors : int
+        The number of components/factors to retain
+    details : pd.DataFrame
+        The eigenvalues and CNG indices of the dataset
+    """
+    data = corr(data.values)
+    if model == "factors":
+        data -= np.linalg.pinv(np.diag(np.diag(np.linalg.pinv(data))))
+        # TODO: Should this line be here?
+        data = covariance_to_correlation(data)
```

Comment:
This line comes from the [`eigenComputes` function](https://rdrr.io/cran/nFactors/src/R/eigenComputes.r). When the data passed in are correlations and `cor == TRUE`, it calls `eigen(cov2cor(corFa(x)))`. However, it _doesn't_ do this if the variables/observations themselves are passed, in which case it calls `eigen(corFA(cov(x)))`. This produces very different results! I think this is only needed if covariances are passed in rather than correlations, but since this implementation calculates the correlations I think this line should be removed. Agreed?

Reply:
So, are we assuming the input value here will always been `eigenForm == "data"`? If so, I think you're right that this should be removed, but I may be misunderstanding.

Reply:
Yeah, I think you're right. I was getting confused by the case where `dataType == correlation && cor == FALSE`, but I think in that case the data passed in should actually be a covariance matrix, not a correlation matrix.